src/posit/connect/users.py

```diff
@@ -117,21 +159,188 @@ class Users(Resources):
+        password : str, optional
+            Applies when server setting 'Authentication.Provider="password"'. Cannot be set when `user_must_set_password` is `True`. Default is `None`.
+        user_must_set_password : bool, optional
+            If `True`, the user is prompted to set their password on first login. When `False`, the `password` parameter is used. Default is `False`. Applies when server setting 'Authentication.Provider="password"'.
+        email : str, optional
+            The user's email address. Default is `None`.
+        first_name : str, optional
+            The user's first name. Default is `None`.
+        last_name : str, optional
+            The user's last name. Default is `None`.
+        user_role : Literal["administrator", "publisher", "viewer"], optional
+            The user role. Default is `None`. Options are `'administrator'`, `'publisher'`, `'viewer'`. Falls back to server setting 'Authorization.DefaultUserRole'.
+        unique_id : str, optional
+            Default is `None`. Required when server is configured with SAML or OAuth2 (non-Google) authentication. Applies when server setting `ProxyAuth.UniqueIdHeader` is set.
+
+        Returns
+        -------
+        User
+            The newly created user.
+
+        Examples
+        --------
+        Create a user with a predefined password:
+
+        >>> user = client.create(
+        ...     username="jdoe",
+        ...     email="jdoe@example.com",
+        ...     first_name="John",
+        ...     last_name="Doe",
+        ...     password="s3cur3p@ssword",
+        ...     user_role="viewer",
+        ... )
+
+        Create a user who must set their own password:
+
+        >>> user = client.create(
+        ...     username="jdoe",
+        ...     email="jdoe@example.com",
+        ...     first_name="John",
+        ...     last_name="Doe",
+        ...     user_must_set_password=True,
+        ...     user_role="viewer",
+        ... )
+        """
+        ...
+
+    @overload
+    def create(self, **attributes) -> User:
+        """
+        Create a user with the specified attributes.
+
+        Parameters
+        ----------
+        **attributes
+            Arbitrary keyword arguments representing user attributes.
+
+        Returns
+        -------
+        User
+            The newly created user.
+        """
+        ...
+
+    def create(self, **attributes) -> User:
+        """
+        Create a user.
+
+        Parameters
+        ----------
+        **attributes
+            Arbitrary keyword arguments representing user attributes.
+
+        Returns
+        -------
+        User
+            The newly created user.
+        """
+        # todo - use the 'context' module to inspect the 'authentication' object and route to POST (local) or PUT (remote).
+        url = self.params.url + "v1/users"
+        response = self.params.session.post(url, json=attributes)
+        return User(self.params, **response.json())
```

Comment:
Remind me: what's the purpose of each successive `@overload`? I'm sure it's written out somewhere.

Reply:
I get that some of the overloads are documentation — it just is non-obvious to me reading this how the two I've highlighted here interact.

Reply:
"How they interact" may be the wrong way to think about it. The overload definitions "[are for the benefit of the type checker only](https://docs.python.org/3/library/typing.html#typing.overload)." The final overload definition says you can call this method within any keyword arguments (`**attributes`). 

I've chosen this to say that you can call this method with any arguments, and they will be passed as the request body. This covers cases where some versions of Connect accept a request body that is different from those defined in the previous overload method definitions. Ideally, we have correct overload definitions for every version of Connect. But, this works as a catch-all fallback in cases where we are unaware.

Would it be helpful to include this information as part of the docstring?